pkg/k8s/service_cache.go

```diff
@@ -81,9 +81,12 @@ type ServiceCache struct {
 
 	// mutex protects the maps below including the concurrent access of each
 	// value.
-	mutex     lock.RWMutex
-	services  map[ServiceID]*Service
-	endpoints map[ServiceID]*Endpoints
+	mutex    lock.RWMutex
+	services map[ServiceID]*Service
+	// endpoints maps a service to a map of endpointSlices. In case the cluster
```

Comment:
Is it allowed by k8s to create an endpoint slice with an empty name?

Reply:
No, I've changed this to make it clear.